tests/gold_tests/timeout/quic_no_activity_timeout.test.py

```diff
@@ -0,0 +1,146 @@
+        Test_quic_no_activity_timeout.server_counter += 1
+        self._server = server
+
+    def _configure_traffic_server(self, tr: 'TestRun'):
+        """Configure Traffic Server.
+
+        :param tr: The TestRun object to associate the ts process with.
+        """
+        ts = tr.MakeATSProcess(f"ts-{Test_quic_no_activity_timeout.ts_counter}", enable_quic=True, enable_tls=True)
+
+        Test_quic_no_activity_timeout.ts_counter += 1
+        self._ts = ts
+        self._ts.addSSLfile("ssl/cert.crt")
+        self._ts.addSSLfile("ssl/private-key.key")
+        self._ts.Disk.records_config.update({
+            'proxy.config.diags.debug.enabled': 1,
+            'proxy.config.diags.debug.tags': 'ssl|net|v_quic|quic|http|socket|inactivity_cop',
+            'proxy.config.quic.no_activity_timeout_in': self.no_activity_timeout_in,
+            'proxy.config.quic.qlog.file_base': f'log/qlog_{Test_quic_no_activity_timeout.server_counter}',
+        })
+
+        if self.extra_recs:
+            self._ts.Disk.records_config.update(self.extra_recs)
+
+        self._ts.Disk.remap_config.AddLine(f'map / http://127.0.0.1:{self._server.Variables.http_port}')
+        self._ts.Disk.ssl_multicert_config.AddLine(
+            f'dest_ip=* ssl_cert_name={self._ts.Variables.SSLDir}/cert.crt ssl_key_name={self._ts.Variables.SSLDir}/private-key.key'
+        )
+
+    def run(self, check_for_max_idle_timeout=False):
+        """Run the test."""
+        tr = Test.AddTestRun(self.name)
+        self._configure_server(tr)
+        self._configure_traffic_server(tr)
+
+        tr.Processes.Default.StartBefore(self._server)
+        tr.Processes.Default.StartBefore(self._ts)
+
+        tr.AddVerifierClientProcess(
+            f'client-{Test_quic_no_activity_timeout.client_counter}',
+            self.replay_file,
+            http3_ports=[self._ts.Variables.ssl_port],
+            keys=self.replay_keys)
+        Test_quic_no_activity_timeout.client_counter += 1
+
+        if check_for_max_idle_timeout:
+            tr.Processes.Default.ReturnCode = 1  # timeout
+            self._ts.Disk.traffic_out.All = Testers.IncludesExpression(
+                "QUIC Idle timeout detected", "We should detect the timeout.")
+        else:
+            tr.Processes.Default.ReturnCode = 0
+
+        if self.gold_file:
+            tr.Processes.Default.Streams.all = self.gold_file
+
+# Tests start.
+
+
+test0 = Test_quic_no_activity_timeout(
+    "Test ts.quic.no_activity_timeout_in(quic max_idle_timeout), no delays",
+    no_activity_timeout_in=0,  # no timeout `max_idle_timeout`
+    replay_keys="nodelays")
+test0.run()
+
+test1 = Test_quic_no_activity_timeout(
+    "Test ts.quic.no_activity_timeout_in(quic max_idle_timeout) with a 5s delay",
+    no_activity_timeout_in=3000,  # 3s `max_idle_timeout`
+    replay_keys="delay5s",
+    gold_file="gold/quic_no_activity_timeout.gold")
+test1.run(check_for_max_idle_timeout=True)
+
+test2 = Test_quic_no_activity_timeout(
```

Comment:
What is this test for? Looks like checking the same thing with shorter times.

Reply:
just removed this one, I think once I have a fix for the crash issue I can enable the commented one.